clang/lib/StaticAnalyzer/Core/LoopUnrolling.cpp

```diff
@@ -226,6 +226,21 @@ static bool isPossiblyEscaped(ExplodedNode *N, const DeclRefExpr *DR) {
           return false;
       }
     }
+
+    if (const SwitchStmt *SS = dyn_cast<SwitchStmt>(S)) {
+      if (const CompoundStmt *CST = dyn_cast<CompoundStmt>(SS->getBody())) {
+        for (const Stmt *CB : CST->body()) {
+          if (const DeclStmt *DST = dyn_cast<DeclStmt>(CB)) {
+            for (const Decl *D : DST->decls()) {
+              // Once we reach the declaration of the VD we can return.
+              if (D->getCanonicalDecl() == VD)
+                return false;
+            }
+          }
```

Comment:
This is duplicated from the previous check. Consider refactoring the code to hoist and name the common functionality.

Reply:
Sure, I can do that.